modules/abstract-substrate/src/abstractSubstrateCoin.ts

```diff
@@ -110,8 +119,31 @@ export class SubstrateCoin extends BaseCoin {
   }
 
   /** @inheritDoc **/
-  isWalletAddress(params: VerifyAddressOptions): Promise<boolean> {
-    throw new MethodNotImplementedError();
+  async isWalletAddress(params: SubstrateVerifyAddressOptions): Promise<boolean> {
+    const { address, keychains } = params;
+
+    const index = Number(params.index ?? params.coinSpecific?.index);
+    if (isNaN(index) || index < 0) {
+      throw new Error('Invalid or missing index. index must be a non-negative number.');
+    }
+
+    const tssParams: TssVerifyAddressOptions = {
+      address,
+      keychains: keychains as TssVerifyAddressOptions['keychains'],
+      index,
+    };
+
+    const isValid = await verifyEddsaTssWalletAddress(
+      tssParams,
+      (addr) => this.isValidAddress(addr),
+      (pubKey) => this.getAddressFromPublicKey(pubKey)
+    );
+
+    if (!isValid) {
+      throw new Error(`Address verification failed: address ${address} is not a wallet address at index ${index}`);
```

Comment:
just do, to keep it consistent, no need to add index validation as derived address will anyway not match
```  async isWalletAddress(params: TssVerifyAddressOptions): Promise<boolean> {
    return verifyEddsaTssWalletAddress(
      params,
      (address) => this.isValidAddress(address),
      (publicKey) => this.getAddressFromPublicKey(publicKey)
    );
  }
```